Add Blend Modes and Canonical Show instances for Enumerations

igagen opened the pull request:
Did you have something else in mind for the show implementation?

paf31 commented on src/Graphics/Canvas.purs:
```diff
@@ -482,6 +483,15 @@ foreign import drawImageFull :: forall eff. Context2D -> CanvasImageSource -> Nu
 data PatternRepeat = Repeat | RepeatX | RepeatY | NoRepeat
 
+instance showPatternRepeat :: Show PatternRepeat where
+  show Repeat = "repeat"
```
What I was getting at before was that we expect valid `Show` instances to produce strings which are executable code. So this and the `Show TextAlign` instance should be changed.

I was suggesting the current `show` be turned into `toString` which would be put into a `where` clause somewhere and not exported.

igagen commented:
Ok, I updated the `Show` instances for `TextAlign` and `PatternRepeat`, and `Composite` as well, and replaced the `show` calls with `toString` as a private helper in the `where` clause. By the way, I noticed that a bunch of the newer composite operations are not in `Composite` (e.g. multiply, screen, etc.). I could add those in if you want.

paf31 commented:
This looks great, thanks!

> I could add those in if you want.

That would be good, if you don't mind.

igagen commented:
Ok great, I'll add those as well :)

igagen commented:
Just added the new blend modes (https://www.w3.org/TR/compositing-1/).

paf31 commented:
Fantastic, thank you!
